v22.1/crdb-internal.md

```diff
@@ -578,6 +579,85 @@ WHERE metadata @> '{"db":"movr"}' AND (metadata @> '{"stmtTyp":"TypeDDL"}' OR me
 
 ~~~
 
+#### Detect suboptimal and regressed plans <span class="version-tag">New in v22.1</span>
+
+To detect suboptimal and regressed plans over time you can compare plans for the same query by extracting them from `statistics->'statistics'->'planGists'`.
+
+Suppose you wanted to compare plans of the following query:
+
+{% include_cached copy-clipboard.html %}
+~~~ sql
+SELECT
+  name, count(rides.id) AS sum
+FROM
+  users JOIN rides ON users.id = rides.rider_id
+WHERE
+  rides.start_time BETWEEN '2018-12-31 00:00:00' AND '2020-01-01 00:00:00'
+GROUP BY
+  name
+ORDER BY
+  sum DESC
+LIMIT
+  10;
+~~~
+
+To decode plan gists stored in `statistics->'statistics'->'planGists'`, you use the `crdb_internal.decode_plan_gist` function. The following example shows a query before (2nd row) and after (1st row) you add an [index on the `start_time` column in the `rides` table](apply-statement-performance-rules.html#rule-2-use-the-right-index). You can see the change in the plans for the same query and the reduced number of rows read and latency when you use the index.
```

Comment:
This paragraph is a little hard to parse.  The order of the before and after is reversed from the order a reader would normally expect.  I was able to puzzle it out but perhaps this could be simplified to use shorter sentences, something like

"To decode plan gists ..., use the .. function as shown in the following query.

The query output shows the performance impact of adding an [index on the  ...]().  The second row of the output shows the performance before the index was added.  The first row of the output shows the improved performance after the index was added."